cmd/event/list/list.go

```diff
@@ -15,71 +15,79 @@
 package list
 
 import (
+	"context"
 	"fmt"
-
 	"io"
-	"strconv"
 	"text/tabwriter"
 
-
 	"github.com/edgexfoundry-holding/edgex-cli/config"
-	 request "github.com/edgexfoundry-holding/edgex-cli/pkg"
 	"github.com/edgexfoundry-holding/edgex-cli/pkg/utils"
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/coredata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
-var limit int32
+var limit int
+var device string
 
 // NewCommand returns the list device command
 func NewCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "list",
-		Short: "A list of all device services",
-		Long:  `Return all device services sorted by id.`,
+		Short: "A list of Events",
+		Long:  `Return list of Events.`,
 		Args:  cobra.MaximumNArgs(1),
-		RunE: func(cmd *cobra.Command, args []string) (err error){
-			var url string
-			if len(args) > 0 {
-				var limitUrl string
-				device := args[0]
-				if limit > 0 {
-					limitUrl = strconv.FormatInt(int64(limit), 10)
-				} else {
-					limitUrl = strconv.FormatInt(int64(50), 10)
-				}
-				url = config.Conf.Clients["CoreData"].Url() + clients.ApiEventRoute + "/device/" + device + "/" + limitUrl
-			} else {
-				url = config.Conf.Clients["CoreData"].Url() + clients.ApiEventRoute
-			}
+		RunE:  listHandler,
+	}
+	cmd.Flags().IntVarP(&limit, "limit", "l", 50, "Limit number of results")
+	cmd.Flags().StringVarP(&device, "device", "d", "", "Events generated by specific device with given name.")
+	return cmd
+}
 
-			var events []models.Event
-			err = request.Get(url, &events)
-			if err != nil {
-				return
-			}
-			pw := viper.Get("writer").(io.WriteCloser)
-			w := new(tabwriter.Writer)
-			w.Init(pw, 0, 8, 1, '\t', 0)
-			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t\n", "Event ID", "Device", "Origin", "Created", "Modified")
+func listHandler(cmd *cobra.Command, args []string) (err error) {
+	client := coredata.NewEventClient(
+		local.New(config.Conf.Clients["CoreData"].Url() + clients.ApiEventRoute),
+	)
 
-			for _, event := range events{
-				fmt.Fprintf(w, "%s\t%s\t%v\t%v\t%s\t\n",
-					event.ID,
-					event.Device,
-					event.Origin,
-					utils.DisplayDuration(event.Created),
-					utils.DisplayDuration(event.Modified),
-				)
-			}
-			w.Flush()
-			return
-		},
+	var events []models.Event
+	if device != "" {
+		events, err = client.EventsForDevice(context.Background(), device, limit)
+	} else if len(args) > 0 {
+		events, err = getEvent(client, args[0])
```

Comment:
Would you please add a comment saying arg[0] contains event-id and we return a list just to share rendering code path. I like your code organization